src/Native/Unix/System.Native/pal_time.c

```diff
@@ -68,15 +68,13 @@ uint64_t SystemNative_GetTimestampResolution()
 
     return (SecondsToNanoSeconds * (uint64_t)(mtid.denom)) / (uint64_t)(mtid.numer);
 #else
-    struct timespec ts;
-
-    if (clock_getres(CLOCK_MONOTONIC, &ts) != 0)
-    {
-        return 0;
-    }
+    // clock_gettime() returns a result in terms of nanoseconds rather than a count. This
+    // means that we need to either always scale the result by the actual resolution (to
+    // get a count) or we need to say the resolution is in terms of nanoseconds. We prefer
+    // the latter since it allows the highest throughput and should minimize error propagated
+    // to the user.
```

Comment:
I'm not entirely understanding.  Was the old implementation wrong?  Just unnecessary work?

Reply:
On all current Linux systems it was just doing unnecessary work (since Linux currently hardcodes to `tv_sec = 0; tv_nsec = 1`. On a hypothetical system where the result of `clock_getres(CLOCK_MONOTONIC)` isn't hardcoded, it would have been wrong.

This is because `clock_gettime()` doesn't return a value that represents a `count`, it returns a timespec which is explicitly in terms of `seconds + nanoseconds` (this is in contrast to `mach_absolute_time` and the real `QPC` which both return a `count`).

The solution is to either divide the returned nanoseconds by the resolution, thereby giving us a `count`, or to just say the resolution is exactly `1 nanosecond` and to just return the result untouched. The latter is probably better because it allows an overall more correct result given that we are returning and dealing with everything as `int64` (and therefore lose any decimal digits due to rounding).